lib/repl.js

```diff
@@ -269,15 +323,55 @@ function REPLServer(prompt,
           return;
         }
       }
-    }
 
-    // After executing the current expression, store the values of RegExp
-    // predefined properties back in `savedRegExMatches`
-    for (var idx = 1; idx < savedRegExMatches.length; idx += 1) {
-      savedRegExMatches[idx] = RegExp[`$${idx}`];
+      if (awaitPromise && !err) {
+        let sigintListener;
+        pause();
+        let promise = result;
+        if (self.breakEvalOnSigint) {
+          const interrupt = new Promise((resolve, reject) => {
+            sigintListener = () => {
+              reject(new Error('Script execution interrupted.'));
+            };
+            prioritizedSigintQueue.add(sigintListener);
+          });
+          promise = Promise.race([promise, interrupt]);
+        }
+
+        promise.then((result) => {
+          if (self.breakEvalOnSigint) {
```

Comment:
Doesn't need the guard, does it?  The .delete() is a no-op when `sigintListener === undefined`.  And wouldn't `if (sigintListener)` be more appropriate anyway?